api/v1beta2/kustomization_types.go

```diff
@@ -0,0 +1,364 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	"time"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/fluxcd/pkg/apis/kustomize"
+	"github.com/fluxcd/pkg/apis/meta"
+	"github.com/fluxcd/pkg/runtime/dependency"
+)
+
+const (
+	KustomizationKind         = "Kustomization"
+	KustomizationFinalizer    = "finalizers.fluxcd.io"
+	MaxConditionMessageLength = 20000
+	DisabledValue             = "disabled"
+)
+
+// KustomizationSpec defines the desired state of a kustomization.
```

Comment:
On the fence if this can be considered an accurate description, and/or adds any value as is.

I personally feel that while it does declare some desire, there is a lot in the object that I wouldn't call "desired state", but rather "build instructions". Making something like this maybe a bit more fitting / descriptive:

```suggestion
// KustomizationSpec defines the configuration to calculate the desired state from a Source using Kustomize.
```